ultravox/tools/ds_tool/ds_tool.py

```diff
@@ -230,6 +255,14 @@ def main(args: DatasetToolArgs):
         data_dict[split] = args.task.map_split(
             ds_split, args.num_workers, args.writer_batch_size, args.exclude_fields
         )
+        # Filter out samples where column_to_check is empty after text_proc.format_asr_text
+        if isinstance(args.task, TextGenerationTask) and args.column_to_check_filter:
```

Comment:
I'd probably push this into `TextGenerationTask.map_split` to encapsulate this logic in one place. Also, will `column_to_check-filter` ever be different than `new_column_name`? ie could you just check the output of the chat completion?

Reply:
column_to_check_filter will always be different than new_column_name. The idea is if the text we pass into chat completion is empty, our chat completion will respond with something like "No text has been provided... etc", which is a row we shoudl throw out.

Reply:
ah ok I see what you're doing now. Maybe the best option here is to make the template raise an error if the data is ill-formed, e.g., https://stackoverflow.com/questions/21778252/how-to-raise-an-exception-in-a-jinja2-macro? Then you can push this logic entirely into the template.